1612 case aggregates frontend

jtimpe opened the pull request:
## Summary of Changes
Pull request closes #1612 

* Implements the frontend for the `DataFileSummary.case_aggregates` added in #2612 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
2. Navigate to Data Files page and submit a data file for the selected reporting month/year
3. Navigate to Submission History and verify the Month/Cases Without Errors/Cases With Errors/Records Unable to Process columns.


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x]  Case data populates table per Figma spec
+ [x] For files with "rejected" or equivalent status, cells show "N/A" and have custom styling
+ [x] Testing Checklist has been run and all tests pass
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [x] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

Commit:
saving state real quick

andrew-jameson committed:
Merge branch 'develop' into feat/1613-minimal-retouch

Commit:
finishing merge with latest

Commit:
Missed old test script

Commit:
Added new test, more cleanup

Commit:
Updating unit tests in DFS, preparing for 1610

Commit:
Merging in Jan's 1610 code for parserError useful-ness

Commit:
Revert "Merging in Jan's 1610 code for parserError useful-ness"

This reverts commit c5796da69d0e9a6d356057550378d536e2be5f8b.

Commit:
update to test to use dfs fixture

Commit:
saving state before new 1610 merge

Commit:
Merged develop and 1610 into 1613 latest

Commit:
Resolving merge conflicts with 1610.

Commit:
Linting changes and comparing to 1610

Commit:
Some unit test linting but inherited 1610 issues

Commit:
Re-ordering job to run tests vs lint first.

Commit:
Updates to linting and unit tests.

Commit:
Merge branch 'feature/1610-parser-error-generator' into feat/1613-minimal-retouch

Commit:
Fixing linting.

andrew-jameson committed:
Update tdrs-backend/setup.cfg

jtimpe committed:
add adminUrl to deployment cypress overrides

Commit:
updates per PR.

Commit:
Excluding trailers for rejection

Commit:
Merge branch 'feature/1610-parser-error-generator' into feat/1613-minimal-retouch

Commit:
VSCode merge resolution is garbage.

Commit:
Fixing precheck for not implemented types

andrew-jameson committed:
Adding "beta" banners to relevant error report sections (#2522)

* Update views.py

* Update views.py

* Update SubmissionHistory.jsx

* Update SubmissionHistory.test.js

* Apply suggestions from code review

Co-authored-by: Miles Reiter <mreiter@goraft.tech>

* lint fixes

---------

Co-authored-by: Miles Reiter <mreiter@goraft.tech>
Co-authored-by: Alex P <63075587+ADPennington@users.noreply.github.com>
Co-authored-by: andrew-jameson <ajameson@teamraft.com>

andrew-jameson committed:
Merge branch 'feature/1610-parser-error-generator' into feat/1613-minimal-retouch

Commit:
Updating to error-handle not implemented schema types

Smithh-Co committed:
Create sprint-73-summary.md (#2565)

andrew-jameson committed:
hotfix for large file sizes (#2542)

* hotfix for large file sizes

* apply timeouts/req limits to dev

* filter identity pages from scan

* IGNORE sql injection

---------

Co-authored-by: Jan Timpe <jtimpe@goraft.tech>
Co-authored-by: mo sohani <msohani@goraft.tech>
Co-authored-by: Alex P <63075587+ADPennington@users.noreply.github.com>

jtimpe committed:
updating validation error language

jtimpe committed:
accidentally included coding challenge

jtimpe committed:
rm comments

elipe17 committed:
2550 deactivation email link (#2557)

* - updated nginx buildpack

* - specifying different nginx version

* - Updating changelog

* - added script to update certain apps in cf
- added workflow for each environment in circi

* - fixed base config

* - fixing jobs

* - Updated based on feedback in OH

* - Updating defaults

* - Removing defaults

* - Fixing indent

* - Adding params to config

* test

* test

* - updating work dir

* - Adding checkout

* - adding cf check

* - logging into cf

* - update cf check to install required binary

* - removing unnecessary switch

* - Forcing plugin installation

* - test installing plugin from script also

* - Adding url to email

* - test code for sandbox

* - using my email

* Revert "Merge branch 'update-cf-os' into 2551-deactivation-email-link"

This reverts commit e963b9df48dd1f72ca0c5b192c979bac11851d11, reversing
changes made to cc9cf81e9d76c42f51ffd5e102f6027d3eb5e645.

* Revert "- using my email"

This reverts commit cc9cf81e9d76c42f51ffd5e102f6027d3eb5e645.

* Revert "- test code for sandbox"

This reverts commit 06037747197d17ed8e63b086fcfcf048ecb50dc4.

---------

Co-authored-by: Alex P <63075587+ADPennington@users.noreply.github.com>
Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

lfrohlich committed:
Update README.md (#2577)

Add ATO

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

reitermb committed:
Create 2023, Spring - Testing CSV & Excel-based error reports.md

reitermb committed:
Update README.md

andrew-jameson committed:
Updating deliverable links (#2584)

elipe17 committed:
User viewset not returning/duplicating users (#2573)

* - Fixed issue not allowing pagination to work locally with nginx
- Added ordering to user field to fix duplicates issue

* - fix lint error

* - Removing ID check since we cannot guarantee that the uuid that is generated per test run will be lexigraphically consistent

---------

Co-authored-by: Alex P <63075587+ADPennington@users.noreply.github.com>
Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

elipe17 committed:
Update cf os (#2523)

* - updated nginx buildpack

* - specifying different nginx version

* - Updating changelog

* - added script to update certain apps in cf
- added workflow for each environment in circi

* - fixed base config

* - fixing jobs

* - Updated based on feedback in OH

* - Updating defaults

* - Removing defaults

* - Fixing indent

* - Adding params to config

* test

* test

* - updating work dir

* - Adding checkout

* - adding cf check

* - logging into cf

* - update cf check to install required binary

* - removing unnecessary switch

* - Forcing plugin installation

* - test installing plugin from script also

* - Adding new dependencies

* - adding package

* - fixing broken install

* - fixing libs

* - using correct command

* - gettign correct version of libc

* - trying to upgrade libs

* - testing

* - Updated README and script

* Revert "- Updated README and script"

This reverts commit 92697b3e53d1fd87b8d3e7995abb9093aa26e307.

* - Removed unnecessary circi stuff
- Removed script
- Updated docs to callout updating secondary apps

* - Correct spelling error

---------

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

elipe17 committed:
Item Number Mismatch (#2578)

* - Updated schemas and models to reflect correct item numbers of fields

* - Revert migration

* - Updated header/trailer item numbers

* - Fixed item numbers off by one errors

---------

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

George-Hudson committed:
pipeline filtering (#2538)

* pipeline changes that filter based on paths and branches. circle ci tracks specified branches in order to keep current functionality on HHS side.

* updated syntax to be in line with build-all.yml

* removed comma

* WIP build flow docs

* added Architecture Decision Record for the change to pipeline workflows

* corrected file type of doc to .md

---------

Co-authored-by: George Hudson <ghudson@teamraft.com>
Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

elipe17 committed:
- Updated view to show latest datafiles
- Added admin filter to show newest or all datafile records
- Updated indices to allow easier elastic queries

George-Hudson committed:
Hotfix Devops/2457 path filtering for documentation (#2597)

* pipeline changes that filter based on paths and branches. circle ci tracks specified branches in order to keep current functionality on HHS side.

* updated syntax to be in line with build-all.yml

* removed comma

* WIP build flow docs

* added Architecture Decision Record for the change to pipeline workflows

* corrected file type of doc to .md

* build and test all on PRs even for documentation

---------

Co-authored-by: George Hudson <ghudson@teamraft.com>

Smithh-Co committed:
Create sprint-74-summary.md (#2596)

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

raftmsohani committed:
added URL filters (#2580)

* added URL filters

* allow github to trigger owasp and label deploys (#2601)

Co-authored-by: George Hudson <ghudson@teamraft.com>

---------

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>
Co-authored-by: George Hudson <georgehudson78@gmail.com>
Co-authored-by: George Hudson <ghudson@teamraft.com>

elipe17 committed:
- Updated search indices to have parent FK

elipe17 committed:
- Fix lint errors

elipe17 committed:
- Updated submission tests
- Moved create_datafile to util

elipe17 committed:
- fix lint errors

elipe17 committed:
- removing frontend filtering

elipe17 committed:
- addding datafile to admin model

elipe17 committed:
Revert "- addding datafile to admin model"

This reverts commit 35a6f24c36c3a4c00ddcfc40f20833530b0199f4.

elipe17 committed:
- Fixed issue where datafile FK wasnt populating
- Regenerated migration

elipe17 committed:
- Readding datafile back to admin view now that the error is resolved

elipe17 committed:
- adding datafile back

elipe17 committed:
Revert "- Readding datafile back to admin view now that the error is resolved"

This reverts commit 2807425059fd1b5b355edfb16d30d170cf869d7b.

elipe17 committed:
- Removed unnecessary fields
- Updated dependencies
- Updated filter

elipe17 committed:
- Updated document to include required fields

Smithh-Co committed:
Create sprint-75-summary.md (#2608)

Smithh-Co committed:
Create sprint-76-summary.md (#2609)

Co-authored-by: Andrew <84722778+andrew-jameson@users.noreply.github.com>

elipe17 committed:
- Fixed failing test

elipe17 committed:
- Resolved failing tests

elipe17 committed:
- Corrected merge thrash

elipe17 committed:
- Using randbits to generate pk to get around confilcting sequence pks

elipe17 committed:
Revert "- Using randbits to generate pk to get around confilcting sequence pks"

This reverts commit ac9b0659a62f64c4114c41faf0baa659a92be07c.

elipe17 committed:
- Updating region in fixture instead of factory
- letting django handle transaction for test

elipe17 committed:
- Moved datafile reference to avoid confusion

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into feat/1613-minimal-retouch

Commit:
pushing up incomplete codebase

Commit:
Other unit tests now have passed w/ good error handling

Commit:
Working tests, need to get setup for case aggregates populating via DB

elipe17 committed:
- Updated queries
- Added helper function
- Need to merge in 2579 for queries to work

Commit:
minor improvement to month2int

elipe17 committed:
Merge branch '2579-latest-data' into feat/1613-case-aggregates

elipe17 committed:
- Fixing most merge errors

elipe17 committed:
- Fixing functions

elipe17 committed:
- Updated queries based on generic relation

elipe17 committed:
- Updated queries to count by case number instead of record number

elipe17 committed:
- Added route
- Updated task to create dfs

elipe17 committed:
- updated tests to include dfs

andrew-jameson committed:
Merge branch 'feat/1613-minimal-retouch' into feat/1613-case-aggregates

Commit:
Cleaning up most comments that are no longer necessary and fixed lint issues.

Commit:
making minor updates, still broken tests.

Commit:
updating pipfile.lock and rebuild image resolved test issues

Commit:
Reorganizing tests, still failing in test_parse.py

Commit:
deleted summary file, split into other test scripts.

Commit:
Fixed missing self reference.

Commit:
Linting fixes.

Commit:
Found reference failure in deployed env.

Commit:
Removing extra returns for missing record type.

andrew-jameson committed:
Merge branch 'develop' into feat/1613-minimal-retouch

Commit:
lint fix

andrew-jameson committed:
Merge branch 'feat/1613-minimal-retouch' into feat/1613-case-aggregates

Commit:
Addressed invocation of datafile for failing test

Commit:
lint update for whitespace

Commit:
Intermediary commit, broken test

jtimpe committed:
add reverse relation

jtimpe committed:
frontend summary status

Commit:
new assignemnts in util

elipe17 committed:
- updated rejected query to correctly count objs

elipe17 committed:
Merge branch 'feat/1613-case-aggregates' of https://github.com/raft-tech/TANF-app into feat/1613-case-aggregates

elipe17 committed:
- Fixing most tests

elipe17 committed:
- Fixed user error. Swapped numbers by accident.

elipe17 committed:
- make region None to avoid PK collision

elipe17 committed:
- Fix lint errors

elipe17 committed:
- Updating to avoid warning

jtimpe committed:
add acceptance icon

jtimpe committed:
add reverse relationship to summary model

jtimpe committed:
Merge branch 'feat/1613-case-aggregates' into feature/1610-frontend-parsing-status

andrew-jameson committed:
vscode merge conflict resolution (#2623)

* auto-create the external network

* didn't stage commit properly

* checking diffs, matching 1613.2

* doesn't work in pipeline. must be cached local

* re-commenting in unit test

* lint failures fixed

---------

Co-authored-by: andrew-jameson <ajameson@teamraft.com>

jtimpe committed:
add a pending status

jtimpe committed:
add acceptance status tests

jtimpe committed:
update warning icon

jtimpe committed:
rm old assert

jtimpe committed:
update rejected icon

jtimpe committed:
add case aggregates to submission history

Commit:
url change per me, want pipeline to run e2e

jtimpe committed:
update case aggregate columns

jtimpe committed:
update col names

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

Commit:
Upgraded to querysets, fix PR comments, PE str

Commit:
Merge develop/remote into local

Commit:
missing : not caught locally

Commit:
Merge branch 'develop' into feat/1613-case-aggregates

andrew-jameson committed:
Feat/1613 merge 2 (#2650)

* Create sprint-78-summary.md (#2645)

* Missing/unsaved parser_error for record_type

* removing redundant tests

* Hopefully resolved on unit tests and lint

---------

Co-authored-by: Smithh-Co <121890311+Smithh-Co@users.noreply.github.com>
Co-authored-by: andrew-jameson <ajameson@teamraft.com>

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

Commit:
Merge branch 'feat/1613-case-aggregates' of github.com:raft-tech/TANF-app into feat/1613-case-aggregates

Commit:
icontains

Commit:
tests

Commit:
Changing dict structure per 1612.

Commit:
fixed tests and lint issues, parse is too complex

Commit:
schema_manager replaces schema check

Commit:
Saving state prior to merge-conflict.

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

Commit:
Adopting latest manager, removing old error style.

Commit:
Commented out t6 line during Office hours

Commit:
minor reference update

Commit:
Acclimating to schemaManager

Commit:
lint-fix isinstance

Commit:
syntax mistake with isinstance

andrew-jameson committed:
Apply suggestions from code review

Commit:
reverting search_index merge artifacts.

Commit:
adjusting for removing unused "get-schema()"

Commit:
whitespace lint

Commit:
Feedback from Jan

Commit:
Ensuring tests run/work.

Commit:
Ensure we have leading zero in rptmonthyear.

jtimpe committed:
Merge branch 'feat/1613-case-aggregates' into feature/1610-frontend-parsing-status

jtimpe committed:
Merge branch 'feature/1610-frontend-parsing-status' into 1612-case-aggregates-frontend

Commit:
Minor lint fix for exception logging

jtimpe committed:
fix migration

jtimpe committed:
add backend partially accepted status

jtimpe committed:
add frontend partially accepted status

jtimpe committed:
add partially accepted test

jtimpe committed:
Merge branch 'feature/1610-frontend-parsing-status' into 1612-case-aggregates-frontend

jtimpe committed:
rm accidental commit

jtimpe committed:
rm unused variable

jtimpe committed:
Merge branch 'feature/1610-frontend-parsing-status' into 1612-case-aggregates-frontend

jtimpe committed:
conditional submission history

jtimpe committed:
Merge branch 'feat/1613-case-aggregates' into 1612-case-aggregates-frontend

jtimpe committed:
fix test

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

Commit:
resolving merge conflict problems

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

Commit:
fixing tests from merge conflicts.

elipe17 committed:
- fixed migration reference

Commit:
dumb lint fix

elipe17 committed:
- Fixed render error

elipe17 committed:
Merge branch 'feat/1613-case-aggregates' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

elipe17 committed:
Adding section

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

elipe17 committed:
- Resolved type mismatches causing render failures

elipe17 committed:
- Adding branch coverage

elipe17 committed:
- Fix lint errors

elipe17 committed:
- Fix lint errors

elipe17 commented on tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.test.js:
```diff
@@ -339,4 +337,181 @@ describe('SubmissionHistory', () => {
     expect(screen.queryByText('test6.txt')).not.toBeInTheDocument()
   })
+
+  it.each([
```
I don't love duplicating the test the way I have. I tried creating a table in the argument of it.each like this: `['Pending','Active Case Data'],['Pending', 'Closed Case Data'],` but that caused errors in the components. Any other suggestions?

elipe17 committed:
- Consolodated tetss

jtimpe reviewed (commented):
changes look good to me @elipe17 , thank you! it doesn't technically allow me to approve as an author of the pr

raftmsohani commented on tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx:
```diff
@@ -68,17 +228,33 @@ const SectionSubmissionHistory = ({ section, label, files }) => {
-    <div className="submission-history-section">
+    <div
+      className="submission-history-section usa-table-container--scrollable"
+      style={{ maxWidth: '1400px', margin: '100px' }}
```
The max width here depends on the size of screen, so I suggest going with % instead of absolute value

andrew-jameson committed:
Merge branch 'develop' into feat/1613-case-aggregates

andrew-jameson committed:
reducing line length for lint

ADPennington committed:
Merge branch 'develop' into feat/1613-case-aggregates

elipe17 committed:
- updated maxWidth to be percentage

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

elipe17 committed:
Merge branch 'feat/1613-case-aggregates' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

Commit:
Moving DFS migration into it's own file to avoid conflicts.

ADPennington committed:
Merge branch 'develop' into feat/1613-case-aggregates

andrew-jameson reviewed (approved):
LGTM!

![Screenshot 2023-09-21 at 10 55 24 AM](https://github.com/raft-tech/TANF-app/assets/84722778/a2f99d70-e0a0-4bf0-b2d4-55d8ddbf4d7a)

raftmsohani commented on tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx:
```diff
@@ -68,17 +228,33 @@ const SectionSubmissionHistory = ({ section, label, files }) => {
-    <div className="submission-history-section">
+    <div
+      className="submission-history-section usa-table-container--scrollable"
+      style={{ maxWidth: '100%', margin: '100px' }}
```
Nit: Looks like the table is centered, I think if we align it to the left, it will look nicer.
![Screenshot 2023-09-22 at 9 23 42 AM](https://github.com/raft-tech/TANF-app/assets/97037188/fdd122de-8e2f-4fba-8196-557c8c3c5652)

ADPennington replied:
im still seeing center-alignment. we should double-check this when moved to gov a11y review. dont want to re-deploy yet @reitermb @ttran-hub @jtimpe

raftmsohani reviewed (approved):
LGTM! Only small nit

ADPennington commented:
@reitermb @ttran-hub this will need a11y review. i'm deploying this to qasp environment. please see Mo's comment about alignment ⬆️

jtimpe committed:
Merge branch 'feat/1613-case-aggregates' into 1612-case-aggregates-frontend

jtimpe committed:
rename migrations

jtimpe committed:
left align error report text

jtimpe committed:
github got stuck :(

jtimpe committed:
rm comment

elipe17 committed:
Merge branch 'feat/1613-case-aggregates' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

elipe17 committed:
Merge branch '1612-case-aggregates-frontend' of https://github.com/raft-tech/TANF-app into 1612-case-aggregates-frontend

ADPennington commented:
@jtimpe i need to do more testing but im perplexed by this:

![1612p2](https://github.com/raft-tech/TANF-app/assets/63075587/bd655d8e-4a71-44d5-a310-bd5b40c96dcd)

jtimpe committed:
merge develop into 1612

jtimpe commented:
> @jtimpe i need to do more testing but im perplexed by this:
> 
> ![1612p2](https://user-images.githubusercontent.com/63075587/270457051-bd655d8e-4a71-44d5-a310-bd5b40c96dcd.PNG)

@ADPennington  that is strange, because that's not what I see

![image](https://github.com/raft-tech/TANF-app/assets/111305129/393c7b85-c720-4d42-af9f-2956a569a7ef)

Was `small_correct_file` modified at all for your test?

ADPennington committed:
Merge branch 'develop' into 1612-case-aggregates-frontend

reitermb reviewed (changes requested):
Definitely second fixing the table alignment issue—I dug around and it looks like it's actually a margin problem rather than alignment per se. Removing the margin: 100px on the table div with the class "submission-history-section usa-table-container--scrollable" fixes the issue.

Another a11y issue is a minor error on the blank state of the error reports cell 
![image](https://github.com/raft-tech/TANF-app/assets/28515874/9476b6ba-f9eb-496c-82e0-09f0ac5ec677)

That's something we should be filling in with appropriate text when it doesn't have that download button in it, e.g. 
![image](https://github.com/raft-tech/TANF-app/assets/28515874/c99e3a4e-916f-4235-b73a-22fd46db7a4e)

On a broad QA note I don't seem to be getting accurate cases with/without error counts either, and mine also seem to differ from other screenshots in here despite using the same test files. Ex submitted for Indiana Q1 2023 in QASP:
small_tanf_section1(1).txt is reading 0s across the board while accepted with errors (on two cases according to the error report).

reitermb commented:
> Definitely second fixing the table alignment issue—I dug around and it looks like it's actually a margin problem rather than alignment per se. Removing the margin: 100px on the table div with the class "submission-history-section usa-table-container--scrollable" fixes the issue.
> 
> Another a11y issue is a minor error on the blank state of the error reports cell ![image](https://user-images.githubusercontent.com/28515874/271669931-9476b6ba-f9eb-496c-82e0-09f0ac5ec677.png)
> 
> That's something we should be filling in with appropriate text when it doesn't have that download button in it, e.g. ![image](https://user-images.githubusercontent.com/28515874/271670343-c99e3a4e-916f-4235-b73a-22fd46db7a4e.png)
> 
> On a broad QA note I don't seem to be getting accurate cases with/without error counts either, and mine also seem to differ from other screenshots in here despite using the same test files. Ex submitted for Indiana Q1 2023 in QASP: small_tanf_section1(1).txt is reading 0s across the board while accepted with errors (on two cases according to the error report).

cc @jtimpe @ADPennington @ttran-hub

jtimpe committed:
rm margin, add text for no error report

jtimpe committed:
fix tests

jtimpe committed:
Merge branch 'develop' into 1612-case-aggregates-frontend

jtimpe commented:
> Definitely second fixing the table alignment issue—I dug around and it looks like it's actually a margin problem rather than alignment per se. Removing the margin: 100px on the table div with the class "submission-history-section usa-table-container--scrollable" fixes the issue.
> 
> Another a11y issue is a minor error on the blank state of the error reports cell ![image](https://user-images.githubusercontent.com/28515874/271669931-9476b6ba-f9eb-496c-82e0-09f0ac5ec677.png)
> 
> That's something we should be filling in with appropriate text when it doesn't have that download button in it, e.g. ![image](https://user-images.githubusercontent.com/28515874/271670343-c99e3a4e-916f-4235-b73a-22fd46db7a4e.png)
> 
> On a broad QA note I don't seem to be getting accurate cases with/without error counts either, and mine also seem to differ from other screenshots in here despite using the same test files. Ex submitted for Indiana Q1 2023 in QASP: small_tanf_section1(1).txt is reading 0s across the board while accepted with errors (on two cases according to the error report).

@reitermb good callouts, thank you! i've removed the offending margin and added in "Pending" and "No Errors" states to the error reports column. Regarding the inconsistent record/error counts - this is something we're aware of and will be discussing, but having unique `CASE_NUMBER` values for the records in your test file might help

reitermb commented:
> > Definitely second fixing the table alignment issue—I dug around and it looks like it's actually a margin problem rather than alignment per se. Removing the margin: 100px on the table div with the class "submission-history-section usa-table-container--scrollable" fixes the issue.
> > Another a11y issue is a minor error on the blank state of the error reports cell ![image](https://user-images.githubusercontent.com/28515874/271669931-9476b6ba-f9eb-496c-82e0-09f0ac5ec677.png)
> > That's something we should be filling in with appropriate text when it doesn't have that download button in it, e.g. ![image](https://user-images.githubusercontent.com/28515874/271670343-c99e3a4e-916f-4235-b73a-22fd46db7a4e.png)
> > On a broad QA note I don't seem to be getting accurate cases with/without error counts either, and mine also seem to differ from other screenshots in here despite using the same test files. Ex submitted for Indiana Q1 2023 in QASP: small_tanf_section1(1).txt is reading 0s across the board while accepted with errors (on two cases according to the error report).
> 
> @reitermb good callouts, thank you! i've removed the offending margin and added in "Pending" and "No Errors" states to the error reports column. Regarding the inconsistent record/error counts - this is something we're aware of and will be discussing, but having unique `CASE_NUMBER` values for the records in your test file might help



Sure, the case numbers are 11111111112 and 11111111140. I've also attached the file & error report. 

[2023-Q1-Active Case Data (14).xlsx](https://github.com/raft-tech/TANF-app/files/12771028/2023-Q1-Active.Case.Data.14.xlsx)
[2023.Q1.Active Case Data (2).txt](https://github.com/raft-tech/TANF-app/files/12771027/2023.Q1.Active.Case.Data.2.txt)

With those cell fill and table margin fixes committed we should be good on the a11y side!

reitermb reviewed (approved):
Confirming that the margin/empty cell fixes are looking good a11y wise! Ready for gov review @ttran-hub @ADPennington 
![image](https://github.com/raft-tech/TANF-app/assets/28515874/c0a0ad8f-f00d-4af1-94be-3625e9612c9c)

reitermb commented:
Officially bumping into QASP a11y/dev review post sync with @ttran-hub.

cc @ADPennington 

Nothing that blocks QASP/indicates a compliance issue on this ticket but also noting that we'll be doing some follow-on a11y research on this ticket to confirm that the way screen-readers are interacting with the table rowspans is the ideal behavior for those. If not, it might be an area for a follow-on ticket to add more descriptive text to accompany the table via the table's caption attribute or aria properties.

ttran-hub commented:
> Officially bumping into QASP a11y/dev review post sync with @ttran-hub.
> 
> cc @ADPennington
> 
> Nothing that blocks QASP/indicates a compliance issue on this ticket but also noting that Thomas, Diana, and I will do some follow-on a11y research on this ticket to confirm that the way screen-readers are interacting with the table rowspans is the ideal behavior for those. If not, it might be an area for a follow-on ticket to add more descriptive text to accompany the table via the table's caption attribute or aria properties.

Thanks @reitermb for the added note on the follow-on a11y research on the way screen readers are interacting with the table rowspans. This should not hold up the PR. a11y 🚀 cc @ADPennington

jtimpe committed:
Merge branch 'develop' into 1612-case-aggregates-frontend

ADPennington commented:
@jtimpe @reitermb @ttran-hub  submission history looks great for TANF (note: i referenced other tickets in the screenshot for follow-up issues) but looks a little off for SSP. Please see below. 

![1612p4](https://github.com/raft-tech/TANF-app/assets/63075587/8f955580-490b-4778-bc85-f4cc72766d41)
![1612p3](https://github.com/raft-tech/TANF-app/assets/63075587/906c9739-3c4d-4363-842a-9f1c56ee06cb)

jtimpe committed:
fix ssp case aggregates display

ADPennington reviewed (approved):
thanks for the quick fix @jtimpe 🚀 

![1612p5](https://github.com/raft-tech/TANF-app/assets/63075587/2ff07872-0f1a-45f0-a992-2723707c36fc)
